cpp/src/gandiva/decimal_xlarge.cc

```diff
@@ -103,27 +103,33 @@ static BasicDecimal128 ConvertToDecimal128(int256_t in, bool* overflow) {
   return is_negative ? -result : result;
 }
 
+static int256_t GetScaleMultiplier(int scale) {
+  static constexpr int32_t kMaxScaleMultiplier = 2 * DecimalTypeUtil::kMaxPrecision;
+  DCHECK_GE(scale, 0);
+  DCHECK_LE(scale, kMaxScaleMultiplier);
+
+  // Compute the scale multipliers once.
+  static std::array<int256_t, kMaxScaleMultiplier + 1> multipliers =
```

Comment:
This is not thread safe.

Reply:
The spec says it's thread-safe  https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

Is there an exception for lambda ?

Reply:
I've made this a global variable now.

Reply:
There doesn't seem to be exception for lamda, I didn't know that local static storage was thread safe in C++11, wasn't in C11 and previous C++ versions.